src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -153,77 +169,156 @@ export default class SimidMapCreative extends BaseSimidCreative {
+   * @private 
+   */
+  displayResults_(results, status) {
     if (status == google.maps.places.PlacesServiceStatus.OK) {
       for (let i = 0; i < DEFAULT_LOCATION_NUM_DISPLAYED; i++) {
         this.placeMapMarker_(results[i]);
       }
+      this.closestLocation_ = results[0].geometry.location;
+      this.displayDirections_(this.closestLocation_, this.currentLocation_);
     }
-}
+  }
 
-/**
- * Creates and displays a marker on the map representing a given place.
- * @param {!Object} place A Place Result object.
- * @private 
- */
-placeMapMarker_(place) {
-  const placeIcon = {
-    url: this.markerImage_,
-    scaledSize: new google.maps.Size(MARKER_SIZE, MARKER_SIZE)
-  };
-  const placeMarker = new google.maps.Marker({
-    map: this.map_,
-    position: place.geometry.location,
-    icon: placeIcon
-  });
-  google.maps.event.addListener(placeMarker, 'click', () => {
-    const infowindow = new google.maps.InfoWindow;
-    infowindow.setContent(place.name);
-    infowindow.open(this.map_, this);
-  });
-}
+  /**
+   * Creates and displays a marker on the map representing a given place.
+   * @param {!Object} place A Place Result object.
+   * @private 
+   */
+  placeMapMarker_(place) {
+    const placeIcon = {
+      url: this.markerImage_,
+      scaledSize: new google.maps.Size(MARKER_SIZE, MARKER_SIZE)
+    };
+    const placeMarker = new google.maps.Marker({
+      map: this.map_,
+      position: place.geometry.location,
+      icon: placeIcon
+    });
+    placeMarker.addListener('click', () => {
+      this.closestLocation_ = place.geometry.location;
+      this.displayDirections_(this.currentLocation_, this.closestLocation_);
+    });
+  }
+
+  /**
+   * Displays the route between the starting loaction and destination
+   * based off of the selected travel mode.
+   * @param {!google.maps.LatLng} destination The LatLng coordinates of the destination.
+   * @param {!google.maps.LatLng} startingLocation The LatLng coordinates of the start location.
+   * @private 
+   */
+  displayDirections_(destination, startingLocation) {
+    this.directionsRenderer_.setMap(this.map_);
+    this.calculateRoute_(startingLocation, destination);
+    document.getElementById("travel-method").addEventListener("change", () => {
+      this.calculateRoute_(startingLocation, destination);
+    });
+  }
+
+  /**
+   * Creates a drop down menu where users can choose between
+   * different modes of travel to display directions for.
+   * @private 
+   */
+  createTravelChoices_() {
+    const adContainer = document.getElementById("button_container")
+    const travelMethod = document.createElement('select');
+    travelMethod.setAttribute("id", "travel-method");
+    const walkOption = this.createTravelOption_("Walking");
+    const driveOption = this.createTravelOption_("Driving");
+    const bikeOption = this.createTravelOption_("Bicycling");
+    const transitOption = this.createTravelOption_("Transit");
```

Comment:
is this public transit? Clarify which type of transit this is (trains, boat, subway, bus, etc..)

Reply:
The Maps API has transit as an umbrella term for bus, train, tram, light rail, and subway, so unfortunately I can't include each individual option there-- do you think it would be better for the explanation for transit to go as a comment in the code or to rename the variable as publicTransitOption?

Reply:
I think I would do both - this way the code specifies in the comments but also users understand when they are choosing their preferred mode